lib/puppet/provider/package/gem.rb

```diff
@@ -15,7 +15,52 @@
 
   has_feature :versionable, :install_options, :uninstall_options
 
-  commands :gemcmd => "gem"
+  def self.which_windows_gemcmd(bin)
```

Comment:
I'm not thrilled with what ends up being lightly modified duplicate of which() from util.rb either. 

I thought about extending the original which() to accept a second parameter to specifically exclude 'puppet' paths (or a parameter to be used as a pattern to exclude matching paths) from the each-do-dir loop. 

But I decided to isolate the which_windows_gemcmd() in gem.rb to minimize the change/testing footprint. I reviewed the other executables in the other puppet bin directories, and didn't find any other potential providers (although I anticipate Execs accidentally finding our curl and openssl executables) that would benefit from the new method. I assume that as use of Puppet on Windows increases, this will change.

I'm open to a new, more-generally-usable method with an exclude parameter based upon file_impl.rb.